Source/WebCore/style/StyleScopeRuleSets.cpp

```diff
@@ -370,6 +370,11 @@ const HashSet<AtomString>& ScopeRuleSets::customPropertyNamesInStyleContainerQue
     return *m_customPropertyNamesInStyleContainerQueries;
 }
 
+bool ScopeRuleSets::hasSimpleSelectorsForStyleAttribute() const
+{
+    return !!attributeInvalidationRuleSets(HTMLNames::styleAttr->localName());
+}
```

Comment:
I think this is just `hasSelectorsForStyleAttribute`, it covers any position the style attribute is mentioned in selectors. This is also the correct thing to do considering how it is used.